### PR TITLE
Support esptool in path.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

**Comment by floitsch** [comment]
When packaging Toit for Linux, there is sometimes (for example Arch Linux) already an esptool package which Toit can depend on.

File: tools/firmware.toit [hunk 1]
```diff
@@ -577,14 +577,26 @@ find_esptool_ -> List:
...
+      return esptool
+  // Try to find esptool in PATH.
+  esptool := ["esptool$bin_extension"]
+  catch:
+    pipe.backticks esptool "version"
+    // Succeeded, so just return it.
     return esptool
+  // An exception was thrown.
+  // Try to find esptool.py in PATH.
+  if platform != PLATFORM_WINDOWS:
```
Comment (kasperl) [inline tools/firmware.toit:594]: Don't we need something similar on Windows?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tools/firmware.toit (hunk 1 above)
Comment (floitsch) [reply tools/firmware.toit:594]: I don't thing shebang scripts work on Windows.
Python adds a file association of Python to the launcher, but I don't think that triggers when directly called (even through a shell).
I think on Windows we need a real executable.

**Pull request merged**